src/k8s-engine/helm.js

```diff
@@ -120,46 +86,27 @@ function status(name, namespace) {
  * @param {string} chart The chart to install
  * @param {string} namespace The namespace to install the chart in to
  * @param {boolean} createNamespace If Helm should create the namespace
- * @returns {object} the parsed JSON for a Helm install command
+ * @returns {Promise<Object>} the parsed JSON for a Helm install command
  */
-function install(name, chart, namespace, createNamespace) {
-  return new Promise((resolve, reject) => {
-    if (name === undefined) {
-      reject('name required to install');
-    }
-    if (chart === undefined) {
-      reject('chart required to install');
-    }
-
-    let dta = '';
-    let err = '';
-    const args = ['install', name, chart, '--kube-context', 'rancher-desktop', '-o', 'json', '--wait'];
-    if (namespace !== undefined) {
-      args.push('--namespace', namespace);
-    }
-    if (createNamespace) {
-      args.push('--create-namespace');
-    }
-
-    // TODO: There is a lot of repeated code in this file. It could be simplified.
-    const bat = spawn(resources.executable('/bin/helm'), args);
-
-    bat.stdout.on('data', data => {
-      dta += data.toString();
-    });
-
-    bat.stderr.on('data', data => {
-      err += data.toString();
-    });
-
-    bat.on('exit', code => {
-      if (code === 0) {
-        resolve(JSON.parse(dta));
-      } else {
-        reject('Failed to list resource: ' + err);
-      }
-    });
-  });
+async function install(name, chart, namespace, createNamespace) {
+  if (name === undefined) {
+    throw new Error('name required to install');
+  }
+  if (chart === undefined) {
+    throw new Error('chart required to install');
+  }
+  const options = { output: 'json', 'kube-context': 'rancher-desktop', wait: undefined };
+  if (namespace !== undefined) {
+    options.namespace = namespace;
+  }
+  if (createNamespace) {
+    options['create-namespace'] = undefined;
+  }
+  try {
+    return await exec(options, 'install', name, chart);
+  } catch (err) {
+    throw new Error(`Failed to install chart: ${err}`);
```

Comment:
Like before, I think we should include the name (and namespace) of an object that we fail to process in the error message.